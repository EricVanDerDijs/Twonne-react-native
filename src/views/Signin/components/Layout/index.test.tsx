import React from 'react';
import { SigninLayout } from './index';
import ShallowRenderer from 'react-test-renderer/shallow';

const shallow = ShallowRenderer.createRenderer();

function mockHandler(): void {
  // tslint:disable-next-line
  console.debug('test text')
}

const initialValues = {
  username: 'eric',
  password: '1234',
}

it('Renders SigninLayout without crashing', () => {
  shallow.render((
    <SigninLayout
      initialValues={initialValues}
      handleSignin={mockHandler}
    />
  ));
});
