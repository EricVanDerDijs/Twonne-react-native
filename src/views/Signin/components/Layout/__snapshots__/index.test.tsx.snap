// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders SigninLayout correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#1A659E",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 35,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "ExpletusSans-BoldItalic",
          "fontSize": 40,
          "textShadowColor": "#FF5454",
          "textShadowOffset": Object {
            "height": 2.3,
            "width": -2.3,
          },
          "textShadowRadius": 1,
        }
      }
    >
      Twonne
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 65,
        "padding": 15,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomWidth": 2,
          "borderColor": "#FF5454",
          "borderLeftWidth": 0,
          "borderRadius": 10,
          "borderRightWidth": 0,
          "marginBottom": 20,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 16,
          }
        }
      >
        username
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 6,
            "color": "#181528",
            "fontSize": 20,
          }
        }
        textContentType="username"
        underlineColorAndroid="transparent"
        value="eric"
      />
    </View>
    <View
      style={
        Object {
          "borderBottomWidth": 2,
          "borderColor": "#FF5454",
          "borderLeftWidth": 0,
          "borderRadius": 10,
          "borderRightWidth": 0,
          "marginBottom": 20,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 16,
          }
        }
      >
        password
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 6,
            "color": "#181528",
            "fontSize": 20,
          }
        }
        textContentType="password"
        underlineColorAndroid="transparent"
        value="eric"
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FF5454",
          "borderRadius": 10,
          "height": 50,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={null}
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Signin
        </Text>
      </View>
    </View>
  </View>
</View>
`;
